unixfs/io/bufdagreader.go

```diff
@@ -0,0 +1,41 @@
+package io
+
+import (
+	"bytes"
+	"context"
+	"io"
+)
+
+type bufDagReader struct {
```

Comment:
Why don't we go ahead and replace all of the `NewRNSC` stuff with this now? It seems weird that they both serve roughly the same purpose.